### ISSUE TITLE
Cannot export model, please help.

### ISSUE THREAD

**Issue by kunyiwang** [aruberts/TabTransformerTF#7]
def model_fttransformer_wetlab(repeat_times = 1, batch_size = 5, fnum = 32):
      res = []
  
      for j in range(repeat_times):
  
      feature_num = 256
      data_all_copy = data_projected_vector.copy(deep=True)
      feature = data_all_copy.iloc[:,0:feature_num]
      feature = feature.values
  
      X = feature
  
      pca = PCA(n_components=fnum)
      principalComponents = pca.fit_transform(X)
      data_pca = pd.DataFrame(principalComponents)
      #data_pca = pd.DataFrame(feature)
      data_pca['label'] = data_all_copy['yield_number'].tolist()
      #train_data, test_data = train_test_split(data_pca, test_size=2112/6067, shuffle=False)
      train_data, test_data = train_test_split(data_pca, test_size=100/4055, shuffle=False)
      X_train, X_val = train_test_split(train_data, test_size=0.2)
  
      NUMERIC_FEATURES = list(range(fnum))
  
      sc = StandardScaler()
      X_train.loc[:, NUMERIC_FEATURES] = sc.fit_transform(X_train[NUMERIC_FEATURES])
      X_val.loc[:, NUMERIC_FEATURES] = sc.transform(X_val[NUMERIC_FEATURES])
      test_data.loc[:, NUMERIC_FEATURES] = sc.transform(test_data[NUMERIC_FEATURES])
  
      FEATURES = NUMERIC_FEATURES
      LABEL = 'label'
  
      # To TF Dataset
      train_dataset = df_to_dataset(X_train[FEATURES + [LABEL]], LABEL, shuffle=True)
      val_dataset = df_to_dataset(X_val[FEATURES + [LABEL]], LABEL, shuffle=False)  # No shuffle
      test_dataset = df_to_dataset(test_data[FEATURES + [LABEL]], LABEL, shuffle=False)
  
      ft_linear_encoder = FTTransformerEncoder(
        numerical_features = FEATURES,
        categorical_features = [],
        numerical_data = X_train[FEATURES].values,
        categorical_data =None, # No categorical data
        y = None,
        #numerical_embedding_type='linear',
        numerical_embedding_type='periodic',
        numerical_bins=128,
        embedding_dim=64,
        depth=3,
        heads=6,
        attn_dropout=0.3,
        ff_dropout=0.3,
        explainable=True
      )
  
      # Pass th encoder to the model
      ft_linear_transformer = FTTransformer(
        encoder=ft_linear_encoder,
        out_dim=1,
        out_activation="relu",
      )
  
      LEARNING_RATE = 0.001
      WEIGHT_DECAY = 0.0001
      NUM_EPOCHS = 500
  
      optimizer = tfa.optimizers.AdamW(
          learning_rate=LEARNING_RATE, weight_decay=WEIGHT_DECAY
      )
  
      ft_linear_transformer.compile(
        optimizer = optimizer,
        loss = {"output": tf.keras.losses.MeanSquaredError(name='mse'), "importances": None},
        metrics= {"output": [tf.keras.metrics.RootMeanSquaredError(name='rmse')], "importances": None},
      )
  
      early = EarlyStopping(monitor="val_output_loss", mode="min", patience=16, restore_best_weights=True)
      callback_list = [early]
  
      ft_linear_history = ft_linear_transformer.fit(
        train_dataset,
        epochs=NUM_EPOCHS,
        validation_data=val_dataset,
        callbacks=callback_list
      )
  
      y_pred = ft_linear_transformer.predict(test_dataset)
      df = test_data
      df['y_pred'] = y_pred['output'].ravel()
      #df.columns = ['y_test','y_pred']
  
      #df.sort_values(by='y_pred', inplace=True, ascending=False)
  
      #data_sampled = df[:batch_size]
      #res.append(data_sampled['y_test'].max())
      model_path = "transformer"
      ft_linear_transformer.save(model_path)
  
      scaler_path = "scaler"
      joblib.dump(sc, scaler_path)
  
      return df

data_projected_vector has been pre-defined, all the requirement packages have been installed, I train it as:

    df = model_fttransformer_wetlab(repeat_times = 1, batch_size = 1, fnum = 32)

but got error:

<img width="1185" alt="Screen Shot 2023-11-18 at 1 29 23 AM" src="https://github.com/aruberts/TabTransformerTF/assets/106349280/f8e9159d-d0e9-4b3c-a779-ba6077a066fa">
